### PR TITLE
cassandra: Fix flaky Cassandra integration tests

### PR DESCRIPTION
**Pull request by yanniszark**
**Description of your changes:**

Fix flaky Cassandra integration tests.

What I think happened is:

The integration test for Cassandra waited for Pods to become Running
before trying to connect to the Cassandra cluster. However, the check
should be that the cluster is Ready, not that the container state is Running.

If we only check for the Running state, we bump into a scenario where:
- The Cassandra container has started (thus is in Running state)
- The Cassandra server has not stated yet

and then tests will fail.

**Which issue is resolved by this Pull Request:**
Resolves https://github.com/rook/rook/issues/7569

[test cassandra]

**Checklist:**

- [x] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).

### CODE REVIEW

File: .github/workflows/integration-test-cassandra-suite.yaml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  pull_request:
+
+jobs:
+  TestCassandraSuite:
+    if: ${{ github.event_name == 'pull_request' && github.ref != 'refs/heads/master' }}
+    runs-on: ubuntu-18.04
+    strategy:
+      fail-fast: false
+      matrix:
+        kubernetes-versions : ['v1.15.12', 'v1.20.2']
```
Comment (travisn) [inline .github/workflows/integration-test-cassandra-suite.yaml:18]: Other tests were just updated to 1.21.0
```suggestion
        kubernetes-versions : ['v1.15.12', 'v1.21.0']
```

File: .github/workflows/integration-test-cassandra-suite.yaml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      if: failure() && contains(github.event.pull_request.labels.*.name, 'debug-cassandra-suite')
+      uses: mxschmitt/action-tmate@v3
+
+  TestCassandraSuite-Master:
+    if: github.ref == 'refs/heads/master' || github.ref == 'refs/tags/release-*'
+    runs-on: ubuntu-18.04
+    strategy:
+      fail-fast: false
+      matrix:
+        kubernetes-versions : ['v1.15.12','v1.16.15','v1.18.15','v1.19.7','v1.20.2']
```
Comment (travisn) [inline .github/workflows/integration-test-cassandra-suite.yaml:69]: See #7599 for similar changes to the other tests
```suggestion
        kubernetes-versions : ['v1.15.12','v1.18.15','v1.20.5','v1.21.0']
```

File: .github/workflows/integration-test-cassandra-suite.yaml [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        kubernetes version: ${{ matrix.kubernetes-versions }}
+        start args: --memory 6g --cpus=2
+        github token: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: check k8s cluster status
+      run: |
+        kubectl cluster-info
+        kubectl get pods -n kube-system
+
+    - name: use local disk
```
Comment (travisn) [inline .github/workflows/integration-test-cassandra-suite.yaml:92]: we don't need the local disk for cassandra, right?

File: tests/framework/utils/k8s_helper_test.go [hunk 4]
```diff
@@ -1,140 +0,0 @@
-/*
-Copyright 2016 The Rook Authors. All rights reserved.
```
Comment (travisn) [inline tests/framework/utils/k8s_helper_test.go:2]: Was this related to cassandra, or you just found dead code?

File: .github/workflows/integration-test-cassandra-suite.yaml [hunk 5]
```diff
@@ -0,0 +1,121 @@
+name: Integration test CassandraSuite
```
Comment (travisn) [inline .github/workflows/integration-test-cassandra-suite.yaml:1]: With this new github action passing, we can disable cassandra tests from Jenkins. Inside this test function `func TestCassandraSuite(t *testing.T)`, we would to skip the test. The easiest might be to add an env var like `SKIP_CASSANDRA_TESTS` [here](https://github.com/rook/rook/blob/master/Jenkinsfile#L235) in the Jenksfile, then the test implementation can skip if that env var is set.

**Review by travisn — CHANGES_REQUESTED** [review]

File: tests/framework/utils/k8s_helper_test.go (hunk 4 above)
Comment (yanniszark) [reply tests/framework/utils/k8s_helper_test.go:2]: Dead code from replacing the kubeconfig loading code.

File: .github/workflows/integration-test-cassandra-suite.yaml (hunk 3 above)
Comment (yanniszark) [reply .github/workflows/integration-test-cassandra-suite.yaml:92]: Yeap!

File: .github/workflows/integration-test-cassandra-suite.yaml (hunk 5 above)
Comment (yanniszark) [reply .github/workflows/integration-test-cassandra-suite.yaml:1]: Done!

File: .github/workflows/integration-test-cassandra-suite.yaml (hunk 3 above)
Comment (travisn) [reply .github/workflows/integration-test-cassandra-suite.yaml:92]: So we can remove this section?

Comment (yanniszark) [reply .github/workflows/integration-test-cassandra-suite.yaml:92]: For some reason I thought I had. My bad!

**Review by travisn — APPROVED** [review]

**Pull request merged**